parsimonious/nodes.py

```diff
@@ -90,6 +90,9 @@ def __str__(self):
 
     def __eq__(self, other):
         """Support by-value deep comparison with other nodes for testing."""
+        if not isinstance(other, Node):
+            return NotImplemented
+
         return (other is not None and
```

Comment:
Since we're doing the isinstance() test above, we can lose the `other is not None` clause.